src/Shared/Helpers/OutputBuilder/OutputBuilder.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.CodeAnalysis.Sarif;
+using System;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    /// <summary>
+    /// Builds the output text based on the format specified
+    /// </summary>
+    public class OutputBuilder : IOutputBuilder
+    {
+        public enum OutputFormat
+        {
+            sarifv1,
+            sarifv2,
+            text // no sarif, just text
+        };
+
+        public OutputFormat? currentOutputFormat { get; private set; } = OutputFormat.text; // default = text
+        IOutputBuilder? currentBuilder;
```

Comment:
This field isn't used in derived classes like SarifOutputBuilder (which uses itself as a builder instead of creating a nested builder).  Recommend removing this.

Reply:
changed the class to a factory